### PR TITLE
Fixes issues with can_hear in regards to AI and expensive view checks.

### PR DESCRIPTION
**Pull request by TheDarkElites**
## About the Pull Request

See title.

Done with help from @checkraisefold 

## Why It's Good For The Game

BUG BAD ME FIX

## Changelog

:cl:
fix: The AI talking through holopad with :h now works correctly again.
spellcheck: 'talks but you cannot hear .' has been replaced with 'says something you cannot hear.'
/:cl:

### CODE REVIEW

**Comment by checkraisefold** [comment]
do not merge yet its broken

File: code/modules/mob/living/carbon/human/memetic_insulation.dm [hunk 1]
```diff
@@ -49,13 +47,13 @@ var/debuff_miniscule = 3
...
-		if(hearable_range == AUDIBLE_RANGE_NONE || !(src in hear(AUDIBLE_RANGE_FULL, origin)))
+		if(hearable_range == AUDIBLE_RANGE_NONE || AUDIBLE_RANGE_FULL < get_dist_euclidian(get_turf(src), get_turf(origin)))
 			return 0
 		if(distance_from_origin <= cut_off)
 			return 100
-		return Clamp((((AUDIBLE_RANGE_FULL - Clamp((distance_from_origin - cut_off)**2, 0, AUDIBLE_RANGE_FULL))/AUDIBLE_RANGE_FULL)  * 100), 0, 100)
+		return clamp((((AUDIBLE_RANGE_FULL - clamp((distance_from_origin - cut_off)**2, 0, AUDIBLE_RANGE_FULL))/AUDIBLE_RANGE_FULL)  * 100), 0, 100)
 	else
-		if((is_concealed ? get_turf(origin) : origin) in hear(hearable_range, get_turf(src))) //get_turf is used as you can still hear stuff even if inside a container or in an inventory
+		if(hearable_range <= get_dist_euclidian(get_turf(src), get_turf(origin)))
```
Comment (checkraisefold) [inline code/modules/mob/living/carbon/human/memetic_insulation.dm]: should be >=

File: code/modules/mob/hear_say.dm [hunk 2]
```diff
@@ -61,12 +61,12 @@
...
 			message = "<b>[message]</b>"
 
-	if(!can_hear(speaker) || get_sound_volume_multiplier() < 0.2)
+	if(!can_hear() || get_sound_volume_multiplier() < 0.2)
 		if(!language || !(language.flags & INNATE)) // INNATE is the flag for audible-emote-language, so we don't want to show an "x talks but you cannot hear them" message if it's set
 			if(speaker == src)
 				to_chat(src, SPAN_WARNING("You cannot hear yourself speak!"))
 			else if(can_see(speaker))
-				to_chat(src, "<span class='name'>[speaker_name]</span>[alt_name] talks but you cannot hear \him.")
+				to_chat(src, "<span class='name'>[speaker_name]</span>[alt_name] says something you cannot hear.")
```
Comment (ivanmixo) [inline code/modules/mob/hear_say.dm:69]:
```suggestion
				to_chat(src, "<span class='name'>[speaker_name]</span>[alt_name] says something you cannot hear [speaker.p_them()].")
```

Comment (checkraisefold) [reply code/modules/mob/hear_say.dm:69]: but you cannot hear*
also why deviate from upstream bay? https://github.com/Baystation12/Baystation12/blob/dev/code/modules/mob/hear_say.dm#L50

Comment (ivanmixo) [reply code/modules/mob/hear_say.dm:69]: Because I do not care about upstream bay.

Comment (TheDarkElites) [reply code/modules/mob/hear_say.dm:69]: This would make the message "John Doe says something you cannot hear them"?

**Comment by TheDarkElites** [comment]
should be good for test merge

**Review by Tsurupeta — APPROVED** [review]

**Comment by ivanmixo** [comment]
me when merge with unadressed review

**Comment by Tsurupeta** [comment]
me when reviewer doesn't put up a change request

**Comment by TheDarkElites** [comment]
Me when the unaddressed review makes a message grammatically incorrect and then refuses to respond when questioned about it in review comments :/

**Pull request merged**